src/runtime-rs/crates/resource/src/volume/hugepage.rs

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{
+    collections::HashMap,
+    fs::File,
+    io::{BufRead, BufReader},
+};
+
+use agent::Storage;
+use anyhow::{anyhow, Context, Result};
+use byte_unit::Byte;
+use kata_sys_util::fs::get_base_name;
+use kata_types::mount::KATA_EPHEMERAL_VOLUME_TYPE;
+
+use crate::share_fs::EPHEMERAL_PATH;
+
+use super::Volume;
+
+const PROC_MOUNTS_FILE: &str = "/proc/mounts";
+
+pub(crate) struct Hugepage {
+    storage: Option<Storage>,
+    mount: oci::Mount,
+}
+
+/// handle hugepage
+impl Hugepage {
+    pub(crate) fn new(
+        m: &oci::Mount,
+        options_map: Option<HashMap<Byte, u64>>,
+        fs_options: Option<Vec<String>>,
+    ) -> Result<Self> {
+        // Create mount option string
+        let option = if let Some(page_size) = get_page_size(fs_options) {
+            let page_size =
+                Byte::from_str(page_size).context("failed to create Byte object from String")?;
+            options_map
+                .context("failed to get opetions map")?
+                .get(&page_size)
+                .map(|size| format!("pagesize={},size={}", page_size.get_bytes(), size))
+        } else {
+            None
+        }
+        .context("failed to get huge page options")?;
+        let mut mount = m.clone();
+        let base_name = get_base_name(m.source.clone())?
+            .into_string()
+            .map_err(|e| anyhow!("{:?}", e))?;
+        // Set the mount source path to a path that resides inside the VM
+        mount.source = format!("{}{}{}", EPHEMERAL_PATH, "/", base_name);
+        // Set the mount type to "bind"
+        mount.r#type = "bind".to_string();
```

Comment:
`const BIND_MOUNT_TYPE: &str = "bind"`

Reply:
I would say even if [specialization for str::to_string()](https://github.com/rust-lang/rust/pull/32586) has landed. For clearity, one should always use `to_owned()` on string literals.

See: https://users.rust-lang.org/t/to-string-vs-to-owned-for-string-literals/1441/6

Reply:
fixed